Riot/ViewController/GroupRoomsViewController.m

```diff
@@ -0,0 +1,602 @@
+    
+    if (self.tableView.dataSource)
+    {
+        [self.tableView reloadData];
+    }
+}
+
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+    return kRiotDesignStatusBarStyle;
+}
+
+// This method is called when the viewcontroller is added or removed from a container view controller.
+- (void)didMoveToParentViewController:(nullable UIViewController *)parent
+{
+    [super didMoveToParentViewController:parent];
+}
+
+- (void)destroy
+{
+    if (kRiotDesignValuesDidChangeThemeNotificationObserver)
+    {
+        [[NSNotificationCenter defaultCenter] removeObserver:kRiotDesignValuesDidChangeThemeNotificationObserver];
+        kRiotDesignValuesDidChangeThemeNotificationObserver = nil;
+    }
+    
+    _group = nil;
+    _mxSession = nil;
+    
+    filteredGroupRooms = nil;
+    
+    groupRooms = nil;
+    
+    [super destroy];
+}
+
+- (void)viewWillAppear:(BOOL)animated
+{
+    [super viewWillAppear:animated];
+    
+    // Screen tracking
+    [[AppDelegate theDelegate] trackScreen:@"GroupDetailsRooms"];
+    
+    if (_group)
+    {
+        // Force refresh
+        [self didUpdateGroupRooms:nil];
+        
+        // Trigger a refresh on the group rooms.
+        [self.mxSession updateGroupRooms:_group success:nil failure:^(NSError *error) {
+            
+            NSLog(@"[GroupRoomsViewController] viewWillAppear: group rooms update failed %@", _group.groupId);
+            
+        }];
+    }
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [super viewWillDisappear:animated];
+    
+    // cancel any pending search
+    [self searchBarCancelButtonClicked:_searchBarView];
+}
+
+- (void)withdrawViewControllerAnimated:(BOOL)animated completion:(void (^)(void))completion
+{
+    // Check whether the current view controller is displayed inside a segmented view controller in order to withdraw the right item
+    if (self.parentViewController && [self.parentViewController isKindOfClass:SegmentedViewController.class])
+    {
+        [((SegmentedViewController*)self.parentViewController) withdrawViewControllerAnimated:animated completion:completion];
+    }
+    else
+    {
+        [super withdrawViewControllerAnimated:animated completion:completion];
+    }
+}
+
+- (void)setGroup:(MXGroup*)group withMatrixSession:(MXSession*)mxSession
+{
+    // Cancel any pending search
+    [self searchBarCancelButtonClicked:_searchBarView];
+    
+    _mxSession = mxSession;
+    
+    [self addMatrixSession:mxSession];
+    
+    [self refreshDisplayWithGroup:group];
+}
+
+#pragma mark -
+
+- (void)registerOnGroupChangeNotifications
+{
+    [self cancelRegistrationOnGroupChangeNotifications];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(didUpdateGroupRooms:) name:kMXSessionDidUpdateGroupRoomsNotification object:self.mxSession];
+}
+
+- (void)cancelRegistrationOnGroupChangeNotifications
+{
+    // Remove any pending observers
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
```

Comment:
That kills the listener on kRiotDesignValuesDidChangeThemeNotification too.

Reply:
agree, I will fix

Reply:
@manuroe after checking it. I disagree, the kRiotDesignValuesDidChangeThemeNotification observer is not remove here. But I will update the code to remove only the listeners on group update

Reply:
ok. I am going to merge the PR as is. Note that there is same behavior in GroupParticipantsViewController and GroupHomeViewController